ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/KnownPeers.hs

```diff
@@ -112,13 +112,6 @@ belowTarget actions
           [jobPeerShare actions policy numPeersToReq (Set.toList selectedForPeerShare)]
       }
 
-    -- If we could peer share except that there are none currently available
-    -- then we return the next wakeup time (if any)
-  | numKnownPeers < targetNumberOfKnownPeers
-  , numPeerShareReqsPossible > 0
-  , Set.null availableForPeerShare
-  = GuardedSkip (EstablishedPeers.minPeerShareTime establishedPeers)
```

Comment:
Could you explain why this is not desirable?

Reply:
As I mentioned in the commit message _any_ peer with a peershare timer can block all peersharing. This meant that a node would only peershare 4 times per hour (15min timeout) even if it was below its targets and had more peers to ask.

Reply:
Makes sense. Probably the intention was to not peer share too often if we tried all the peers. But that's not very useful if we already limit how often we peer share with each peer (do we do that?).

Reply:
We peershare at most once every 15 minutes with each individual peer.
But that only happens when we are bellow our targets, so ideally it should only be needed during churning.